### ISSUE TITLE
Remote hdfs

### ISSUE THREAD

**Issue by justmytwospence** [raghavgautam/tramp-hdfs#3]
How can I access the hdfs file system of a remote server when emacs was started locally?

None of the following work for me:

```
/sshx:user@host|hdfs:user@host:/path/to/dir
/sshx:user@host:~|hdfs:user@host:/path/to/dir
/hdfs:user@host:/path/to/dir
```

It says "looks like a remote host, `hdfs' can only use the local host".

I've confirmed that when I'm sshed into the remote server, I can execute the following successfully:

```
curl -s "http://localhost:17302/webhdfs/v1/path/to/dir/?user.name=user&op=LISTSTATUS"
```

and I have changed the webhdfs-port variable appropriately.

**Comment by raghavgautam** [comment]
@justmytwospence Can you upload the the debugging logs ?
Debugging can be enabled by setting:
(setq tramp-verbose 10)

Did you do ?
(setq webhdfs-port 17302)

**Comment by justmytwospence** [comment]
Yes that is what I set for the port.

I see "contacting host: localhost:17302" or "contacting host: <host>:17302" depending on if I try with "localhost" or the actual "<host>", respectively. It just kind of hangs there.

**Comment by justmytwospence** [comment]
Which way is the correct connecting syntax?

**Comment by raghavgautam** [comment]
The syntax is typically tramp syntax. Just open:
/hdfs:user@host:/path/to/dir

You can't supply port as part of the tramp path, you have to set it it configuration.
For hanging Ctrl-g should take you out of the freeze.

**Comment by justmytwospence** [comment]
Okay, that syntax gives me this:

```
Contacting host: host:17302
Use M-x make-directory RET RET to create the directory and its parents
Contacting host: host:17302 [2 times]
```

**Comment by raghavgautam** [comment]
Debug logs ? You can find them in a buffer whose name will be something like:

```
*debug tramp/hdfs root@node-1*
```

Log itself will look like:

```
17:54:01.283434 tramp-get-file-property (8) # / file-attributes-integer undef
17:54:01.360666 tramp-hdfs-get-url-content (10) # Fetched http://node-1:50070/webhdfs/v1/?user.name=root&op=GETFILESTATUS to get:
{"FileStatus":{"accessTime":0,"blockSize":0,"childrenNum":7,"fileId":16385,"group":"hdfs","length":0,"modificationTime":1446035476787,"owner":"hdfs","pathSuffix":"","permission":"755","replication":0,"st\
oragePolicy":0,"type":"DIRECTORY"}}
17:54:01.361409 tramp-get-connection-property (7) # process-name nil
17:54:01.361600 tramp-set-file-property (8) # / file-attributes-integer (t 0 hdfs hdfs (0 0) (22064 49172 786999 940872) (0 0) 0 drwxr-xr-x nil nil (-1 . 1))
17:54:01.362126 tramp-get-file-property (8) # / file-attributes-integer (t 0 hdfs hdfs (0 0) (22064 49172 786999 940872) (0 0) 0 drwxr-xr-x nil nil (-1 . 1))
17:54:01.364977 tramp-get-file-property (8) # /.ede-ignore file-attributes-integer undef
```

**Comment by raghavgautam** [comment]
Also, use the latest tramp-hdfs.

**Comment by justmytwospence** [comment]
I've noticed now that its not actually hanging, a buffer with the proper mode opens, it is just empty.

Logs for

```
/hdfs:username@hostname:/user/username/model.yaml
```

are below; <username> and <hostname> have been replaced as such.

```
;; GNU Emacs: 24.5.1 Tramp: 2.2.11-24.5 -*- mode: outline; -*-
17:57:35.719977 tramp-get-file-property (8) # /user/username/model.yaml file-attributes-integer undef
17:57:36.234468 tramp-get-file-property (8) # /user/username/model.yaml file-attributes-integer undef
17:57:36.238954 tramp-get-file-property (8) # /user/username/model.yaml file-attributes-integer undef
17:57:36.242326 tramp-get-file-property (8) # /user/username/model.yaml file-attributes-integer undef
17:57:36.244219 tramp-get-file-property (8) # /user/username/model.yaml file-attributes-integer undef
17:57:36.245067 tramp-handle-insert-file-contents (10) #
  backtrace()
  tramp-handle-insert-file-contents("/hdfs:username@hostname:/user/username/model.yaml" t nil nil nil)
  apply(tramp-handle-insert-file-contents ("/hdfs:username@hostname:/user/username/model.yaml" t nil nil nil))
  tramp-hdfs-file-name-handler(insert-file-contents "/hdfs:username@hostname:/user/username/model.yaml" t nil nil nil)
  apply(tramp-hdfs-file-name-handler insert-file-contents ("/hdfs:username@hostname:/user/username/model.yaml" t nil nil nil))
  byte-code("\300\301\215\207" [suppress (apply foreign operation args)] 2)
  byte-code(K\306\211<\203\237
@\307=\203\237\306\310\311!\203\"\312\311\313N@!\206\221\314\315!\2039\316\315!\2044\317\315!\205\221\315 \202\221\320\321!\211\205D\322!)\203P\323\320\321!!\202\221\320\324!\211\205[\322!)\203g\323\320\324!!\202\221\320\325!\211\205r\322!)\203~\323\320\325!!\202\221\326\327!\203\212\323\327!\202\221\330\331!\210\323\332!)\333
\211A@\262\334\335#\210)\336\337\215\211\336=\203\312*\340\341+,BC-./0\306\342\343\217,\210\344+,\"\202\374    \345=\203\373\3061*\346\347+,BC-./0\306\342\343\217,\210\350*\351\"\210\344+,\")\202\374   *\207" [foreign result sf file-name-handler-alist d default-directory nil autoload boundp temporary-file-directory eval standard-value fboundp temp-directory subrp functionp getenv "TEMP" file-directory-p file-name-as-directory "TMP" "TMPDIR" file-exists-p "c:/temp" message "Neither `temporary-file-directory' nor `temp-directory' is defined -- using /tmp." "/tmp" load noerror nomessage non-essential (byte-code "\300\301\215\207" [suppress (apply foreign operation args)] 2) 5 "Non-essential received in operation %s" (byte-code     X\205g\306 \307\216\203;\310X\203;\311\31\313U\203!\314\2026\315U\203+\314\2026\316U\2035\317\2026\320P
#\210   \321Y\205f\322!\203S\313\323\324\325#)\326!\205f\311\327\330\33\"P
$*\207" [level tramp-verbose save-match-data-internal tramp-message-show-message fmt-string arguments match-data ((byte-code "\30\302\"\207" [save-match-data-internal set-match-data evaporate] 3)) 3 apply message 0 "" 1 2 "Warning: " "Tramp: " 4 processp tramp-get-connection-property "vector" nil vectorp tramp-debug-message format "(%d) # " vec-or-proc] 6) ((error)) tramp-run-real-handler suppress 1 "Suppress received in operation %s" tramp-cleanup-connection t v operation args arguments fmt-string level vec-or-proc tramp-message-show-message] 6)
  tramp-file-name-handler(insert-file-contents "/hdfs:username@hostname:/user/username/model.yaml" t nil nil nil)
  insert-file-contents("/hdfs:username@hostname:/user/username/model.yaml" t)
  #[0 "\302\303\300\302\")\207" ["/hdfs:username@hostname:/user/username/model.yaml" inhibit-read-only t insert-file-contents] 3 "

(fn)"]()
  funcall(#[0 "\302\303\300\302\")\207" ["/hdfs:username@hostname:/user/username/model.yaml" inhibit-read-only t insert-file-contents] 3 "

(fn)"])
  find-file-noselect-1(#<buffer model.yaml> "/hdfs:username@hostname:/user/username/model.yaml" nil nil "/hdfs:username@hostname:/user/username/model.yaml" nil)
  find-file-noselect("/hdfs:username@hostname:/user/username/model.yaml" nil nil t)
  find-file("/hdfs:username@hostname:/user/username/model.yaml" t)
  call-interactively(find-file nil nil)
  command-execute(find-file)
17:57:36.245192 tramp-handle-insert-file-contents (1) # File error: File `/hdfs:username@hostname:/user/username/model.yaml' not found on remote host
17:57:36.245753 tramp-get-file-property (8) # /user/username/model.yaml file-attributes-integer undef
17:57:36.246714 tramp-get-file-property (8) # /user/username/model.yaml file-attributes-integer undef
17:57:36.247777 tramp-get-file-property (8) # /user/username/model.yaml file-attributes-integer undef
17:57:36.248630 tramp-get-file-property (8) # /user/username file-attributes-integer undef
17:57:36.249598 tramp-get-file-property (8) # /user/username/ file-attributes-integer undef
17:57:37.260333 tramp-get-file-property (8) # /user/username/model.yaml file-attributes-integer undef
17:57:37.275768 tramp-get-file-property (8) # /user/username/ file-attributes-integer undef
17:57:37.277265 tramp-get-file-property (8) # /user/username file-attributes-integer undef
17:57:37.327880 tramp-get-file-property (8) # /user/username/.projectile file-attributes-integer undef
17:57:37.329402 tramp-get-file-property (8) # /user/.projectile file-attributes-integer undef
17:57:37.330810 tramp-get-file-property (8) # /.projectile file-attributes-integer undef
17:57:37.332218 tramp-get-file-property (8) # /user/username/.git file-attributes-integer undef
17:57:37.372813 tramp-get-file-property (8) # /user/.git file-attributes-integer undef
17:57:37.374209 tramp-get-file-property (8) # /.git file-attributes-integer undef
17:57:37.375542 tramp-get-file-property (8) # /user/username/.hg file-attributes-integer undef
17:57:37.376890 tramp-get-file-property (8) # /user/.hg file-attributes-integer undef
17:57:37.378299 tramp-get-file-property (8) # /.hg file-attributes-integer undef
17:57:37.379605 tramp-get-file-property (8) # /user/username/.fslckout file-attributes-integer undef
17:57:37.380898 tramp-get-file-property (8) # /user/.fslckout file-attributes-integer undef
17:57:37.382272 tramp-get-file-property (8) # /.fslckout file-attributes-integer undef
17:57:37.383585 tramp-get-file-property (8) # /user/username/.bzr file-attributes-integer undef
17:57:37.384875 tramp-get-file-property (8) # /user/.bzr file-attributes-integer undef
17:57:37.386176 tramp-get-file-property (8) # /.bzr file-attributes-integer undef
17:57:37.387449 tramp-get-file-property (8) # /user/username/_darcs file-attributes-integer undef
17:57:37.388775 tramp-get-file-property (8) # /user/_darcs file-attributes-integer undef
17:57:37.390099 tramp-get-file-property (8) # /_darcs file-attributes-integer undef
17:57:37.392894 tramp-get-file-property (8) # /user/username/rebar.config file-attributes-integer undef
17:57:37.394158 tramp-get-file-property (8) # /user/username/project.clj file-attributes-integer undef
17:57:37.395421 tramp-get-file-property (8) # /user/username/build.boot file-attributes-integer undef
17:57:37.396681 tramp-get-file-property (8) # /user/username/SConstruct file-attributes-integer undef
17:57:37.397930 tramp-get-file-property (8) # /user/username/pom.xml file-attributes-integer undef
17:57:37.399178 tramp-get-file-property (8) # /user/username/build.sbt file-attributes-integer undef
17:57:37.400485 tramp-get-file-property (8) # /user/username/gradlew file-attributes-integer undef
17:57:37.401765 tramp-get-file-property (8) # /user/username/build.gradle file-attributes-integer undef
17:57:37.403102 tramp-get-file-property (8) # /user/username/Gemfile file-attributes-integer undef
17:57:37.404471 tramp-get-file-property (8) # /user/username/requirements.txt file-attributes-integer undef
17:57:37.405679 tramp-get-file-property (8) # /user/username/setup.py file-attributes-integer undef
17:57:37.406913 tramp-get-file-property (8) # /user/username/tox.ini file-attributes-integer undef
17:57:37.408138 tramp-get-file-property (8) # /user/username/package.json file-attributes-integer undef
17:57:37.409372 tramp-get-file-property (8) # /user/username/gulpfile.js file-attributes-integer undef
17:57:37.410583 tramp-get-file-property (8) # /user/username/Gruntfile.js file-attributes-integer undef
17:57:37.411856 tramp-get-file-property (8) # /user/username/bower.json file-attributes-integer undef
17:57:37.413188 tramp-get-file-property (8) # /user/username/composer.json file-attributes-integer undef
17:57:37.414644 tramp-get-file-property (8) # /user/username/Cargo.toml file-attributes-integer undef
17:57:37.416000 tramp-get-file-property (8) # /user/username/mix.exs file-attributes-integer undef
17:57:37.417307 tramp-get-file-property (8) # /user/username/stack.yaml file-attributes-integer undef
17:57:37.418675 tramp-get-file-property (8) # /user/rebar.config file-attributes-integer undef
17:57:37.419941 tramp-get-file-property (8) # /user/project.clj file-attributes-integer undef
17:57:37.421256 tramp-get-file-property (8) # /user/build.boot file-attributes-integer undef
17:57:37.422516 tramp-get-file-property (8) # /user/SConstruct file-attributes-integer undef
17:57:37.423817 tramp-get-file-property (8) # /user/pom.xml file-attributes-integer undef
17:57:37.425099 tramp-get-file-property (8) # /user/build.sbt file-attributes-integer undef
17:57:37.426628 tramp-get-file-property (8) # /user/gradlew file-attributes-integer undef
17:57:37.427844 tramp-get-file-property (8) # /user/build.gradle file-attributes-integer undef
17:57:37.429073 tramp-get-file-property (8) # /user/Gemfile file-attributes-integer undef
17:57:37.430308 tramp-get-file-property (8) # /user/requirements.txt file-attributes-integer undef
17:57:37.431523 tramp-get-file-property (8) # /user/setup.py file-attributes-integer undef
17:57:37.432778 tramp-get-file-property (8) # /user/tox.ini file-attributes-integer undef
17:57:37.434013 tramp-get-file-property (8) # /user/package.json file-attributes-integer undef
17:57:37.435290 tramp-get-file-property (8) # /user/gulpfile.js file-attributes-integer undef
17:57:37.436604 tramp-get-file-property (8) # /user/Gruntfile.js file-attributes-integer undef
17:57:37.437882 tramp-get-file-property (8) # /user/bower.json file-attributes-integer undef
17:57:37.439242 tramp-get-file-property (8) # /user/composer.json file-attributes-integer undef
17:57:37.440489 tramp-get-file-property (8) # /user/Cargo.toml file-attributes-integer undef
17:57:37.441755 tramp-get-file-property (8) # /user/mix.exs file-attributes-integer undef
17:57:37.443020 tramp-get-file-property (8) # /user/stack.yaml file-attributes-integer undef
17:57:37.444342 tramp-get-file-property (8) # /rebar.config file-attributes-integer undef
17:57:37.445569 tramp-get-file-property (8) # /project.clj file-attributes-integer undef
17:57:37.446866 tramp-get-file-property (8) # /build.boot file-attributes-integer undef
17:57:37.448107 tramp-get-file-property (8) # /SConstruct file-attributes-integer undef
17:57:37.449357 tramp-get-file-property (8) # /pom.xml file-attributes-integer undef
17:57:37.450551 tramp-get-file-property (8) # /build.sbt file-attributes-integer undef
17:57:37.451753 tramp-get-file-property (8) # /gradlew file-attributes-integer undef
17:57:37.452940 tramp-get-file-property (8) # /build.gradle file-attributes-integer undef
17:57:37.454125 tramp-get-file-property (8) # /Gemfile file-attributes-integer undef
17:57:37.502919 tramp-get-file-property (8) # /requirements.txt file-attributes-integer undef
17:57:37.504522 tramp-get-file-property (8) # /setup.py file-attributes-integer undef
17:57:37.505867 tramp-get-file-property (8) # /tox.ini file-attributes-integer undef
17:57:37.507117 tramp-get-file-property (8) # /package.json file-attributes-integer undef
17:57:37.508314 tramp-get-file-property (8) # /gulpfile.js file-attributes-integer undef
17:57:37.509475 tramp-get-file-property (8) # /Gruntfile.js file-attributes-integer undef
17:57:37.510664 tramp-get-file-property (8) # /bower.json file-attributes-integer undef
17:57:37.511886 tramp-get-file-property (8) # /composer.json file-attributes-integer undef
17:57:37.513099 tramp-get-file-property (8) # /Cargo.toml file-attributes-integer undef
17:57:37.514328 tramp-get-file-property (8) # /mix.exs file-attributes-integer undef
17:57:37.515510 tramp-get-file-property (8) # /stack.yaml file-attributes-integer undef
17:57:37.518616 tramp-get-file-property (8) # /user/username/.svn file-attributes-integer undef
17:57:37.520289 tramp-get-file-property (8) # /user/.svn file-attributes-integer undef
17:57:37.521918 tramp-get-file-property (8) # /.svn file-attributes-integer undef
17:57:37.523482 tramp-get-file-property (8) # /user/username/CVS file-attributes-integer undef
17:57:37.524952 tramp-get-file-property (8) # /user/CVS file-attributes-integer undef
17:57:37.526298 tramp-get-file-property (8) # /CVS file-attributes-integer undef
17:57:37.527709 tramp-get-file-property (8) # /user/username/Makefile file-attributes-integer undef
17:57:37.529027 tramp-get-file-property (8) # /user/Makefile file-attributes-integer undef
17:57:37.530296 tramp-get-file-property (8) # /Makefile file-attributes-integer undef
```

**Comment by raghavgautam** [comment]
In the logs it says that it was not able to find the file. Does it exist ?

```
File error: File `/hdfs:username@hostname:/user/username/model.yaml' not found on remote host
```

Also, looks like you are using projectile & tramp, they don't go well together.
https://www.reddit.com/r/emacs/comments/320cvb/projectile_slows_tramp_mode_to_a_crawl_is_there_a/

**Comment by justmytwospence** [comment]
I'm pretty sure the file is there, because this works:

```
curl -s "http://localhost:17302/webhdfs/v1/user/username/model.yaml//?user.name=username&op=LISTSTATUS"
```

```
{"FileStatuses":{"FileStatus":[
{"accessTime":1446079584005,"blockSize":134217728,"childrenNum":0,"fileId":307539850,"group":"username","length":5455,"modificationTime":1445982375636,"owner":"username","pathSuffix":"","permission":"644","replication":3,"storagePolicy":0,"type":"FILE"}
]}}
```

**Comment by raghavgautam** [comment]
I don't see any rest calls in the debug logs. Can you try with latest version ?

**Comment by justmytwospence** [comment]
Using the latest version from melpa (`20151028.2036`):

```
ssh username@hostname
curl -s -s "http://localhost:17302/webhdfs/v1/user/username/model.yaml//?user.name=username&op=LISTSTATUS"
```

```
{"FileStatuses":{"FileStatus":[
{"accessTime":1446079584005,"blockSize":134217728,"childrenNum":0,"fileId":307539850,"group":"username","length":5455,"modificationTime":1445982375636,"owner":"username","pathSuffix":"","permission":"644","replication":3,"storagePolicy":0,"type":"FILE"}
]}}
```

```
(setq tramp-verbose 10)
C-x C-f
/hdfs:username@hostname:/user/username/model.yaml
```

`*Messages*`:

```
Contacting host: hostname:17302
```

`*debug tramp*`:

```
;; GNU Emacs: 24.5.1 Tramp: 2.2.11-24.5 -*- mode: outline; -*-
13:44:35.196800 tramp-get-file-property (8) # /user/username/model.yaml file-attributes-integer undef
13:44:35.286985 tramp-get-file-property (8) # /user/username/model.yaml file-attributes-integer undef
13:44:35.301502 tramp-get-file-property (8) # /user/username/model.yaml file-attributes-integer undef
13:44:35.305257 tramp-get-file-property (8) # /user/username/model.yaml file-attributes-integer undef
13:44:35.343955 tramp-get-file-property (8) # /user/username/model.yaml file-attributes-integer undef
13:44:35.344956 tramp-handle-insert-file-contents (10) # 
  backtrace()
  tramp-handle-insert-file-contents("/hdfs:username@hostname:/user/username/model.yaml" t nil nil nil)
  apply(tramp-handle-insert-file-contents ("/hdfs:username@hostname:/user/username/model.yaml" t nil nil nil))
  tramp-hdfs-file-name-handler(insert-file-contents "/hdfs:username@hostname:/user/username/model.yaml" t nil nil nil)
  apply(tramp-hdfs-file-name-handler insert-file-contents ("/hdfs:username@hostname:/user/username/model.yaml" t nil nil nil))
  byte-code("\300\301\215\207" [suppress (apply foreign operation args)] 2)
  byte-code("K\306\211<\203\237
@\307=\203\237\306\310\311!\203\"\312\311\313N@!\206\221\314\315!\2039\316\315!\2044\317\315!\205\221\315 \202\221\320\321!\211\205D\322!)\203P\323\320\321!!\202\221\320\324!\211\205[\322!)\203g\323\320\324!!\202\221\320\325!\211\205r\322!)\203~\323\320\325!!\202\221\326\327!\203\212\323\327!\202\221\330\331!\210\323\332!)\333
\211A@\262\334\335#\210)\336\337\215\211\336=\203\312*\340\341+,BC-./0\306\342\343\217,\210\344+,\"\202\374    \345=\203\373\3061*\346\347+,BC-./0\306\342\343\217,\210\350*\351\"\210\344+,\")\202\374   *\207" [foreign result sf file-name-handler-alist d default-directory nil autoload boundp temporary-file-directory eval standard-value fboundp temp-directory subrp functionp getenv "TEMP" file-directory-p file-name-as-directory "TMP" "TMPDIR" file-exists-p "c:/temp" message "Neither `temporary-file-directory' nor `temp-directory' is defined -- using /tmp." "/tmp" load noerror nomessage non-essential (byte-code "\300\301\215\207" [suppress (apply foreign operation args)] 2) 5 "Non-essential received in operation %s" (byte-code "  X\205g\306 \307\216\203;\310X\203;\311\312\313U\203!\314\2026\315U\203+\314\2026\316U\2035\317\2026\320P
#\210   \321Y\205f\322!\203S\313\323\324\325#)\326!\205f\311\327\330\331\"P
$*\207" [level tramp-verbose save-match-data-internal tramp-message-show-message fmt-string arguments match-data ((byte-code "\301\302\"\207" [save-match-data-internal set-match-data evaporate] 3)) 3 apply message 0 "" 1 2 "Warning: " "Tramp: " 4 processp tramp-get-connection-property "vector" nil vectorp tramp-debug-message format "(%d) # " vec-or-proc] 6) ((error)) tramp-run-real-handler suppress 1 "Suppress received in operation %s" tramp-cleanup-connection t v operation args arguments fmt-string level vec-or-proc tramp-message-show-message] 6)
  tramp-file-name-handler(insert-file-contents "/hdfs:username@hostname:/user/username/model.yaml" t nil nil nil)
  insert-file-contents("/hdfs:username@hostname:/user/username/model.yaml" t)
  #[0 "\302\303\300\302\")\207" ["/hdfs:username@hostname:/user/username/model.yaml" inhibit-read-only t insert-file-contents] 3 "

(fn)"]()
  funcall(#[0 "\302\303\300\302\")\207" ["/hdfs:username@hostname:/user/username/model.yaml" inhibit-read-only t insert-file-contents] 3 "

(fn)"])
  find-file-noselect-1(#<buffer model.yaml> "/hdfs:username@hostname:/user/username/model.yaml" nil nil "/hdfs:username@hostname:/user/username/model.yaml" nil)
  find-file-noselect("/hdfs:username@hostname:/user/username/model.yaml" nil nil t)
  find-file("/hdfs:username@hostname:/user/username/model.yaml" t)
  call-interactively(find-file nil nil)
  command-execute(find-file)
13:44:35.345073 tramp-handle-insert-file-contents (1) # File error: File `/hdfs:username@hostname:/user/username/model.yaml' not found on remote host
13:44:35.345590 tramp-get-file-property (8) # /user/username/model.yaml file-attributes-integer undef
13:44:35.346514 tramp-get-file-property (8) # /user/username/model.yaml file-attributes-integer undef
13:44:35.347540 tramp-get-file-property (8) # /user/username/model.yaml file-attributes-integer undef
13:44:35.348397 tramp-get-file-property (8) # /user/username file-attributes-integer undef
13:44:35.349354 tramp-get-file-property (8) # /user/username/ file-attributes-integer undef
13:44:36.356846 tramp-get-file-property (8) # /user/username/model.yaml file-attributes-integer undef
13:44:36.372558 tramp-get-file-property (8) # /user/username/ file-attributes-integer undef
13:44:36.374093 tramp-get-file-property (8) # /user/username file-attributes-integer undef
13:44:36.429985 tramp-get-file-property (8) # /user/username/.projectile file-attributes-integer undef
13:44:36.431458 tramp-get-file-property (8) # /user/.projectile file-attributes-integer undef
13:44:36.433113 tramp-get-file-property (8) # /.projectile file-attributes-integer undef
13:44:36.434483 tramp-get-file-property (8) # /user/username/.git file-attributes-integer undef
13:44:36.435908 tramp-get-file-property (8) # /user/.git file-attributes-integer undef
13:44:36.437218 tramp-get-file-property (8) # /.git file-attributes-integer undef
13:44:36.438544 tramp-get-file-property (8) # /user/username/.hg file-attributes-integer undef
13:44:36.439857 tramp-get-file-property (8) # /user/.hg file-attributes-integer undef
13:44:36.441147 tramp-get-file-property (8) # /.hg file-attributes-integer undef
13:44:36.442491 tramp-get-file-property (8) # /user/username/.fslckout file-attributes-integer undef
13:44:36.443813 tramp-get-file-property (8) # /user/.fslckout file-attributes-integer undef
13:44:36.445179 tramp-get-file-property (8) # /.fslckout file-attributes-integer undef
13:44:36.446554 tramp-get-file-property (8) # /user/username/.bzr file-attributes-integer undef
13:44:36.447922 tramp-get-file-property (8) # /user/.bzr file-attributes-integer undef
13:44:36.449344 tramp-get-file-property (8) # /.bzr file-attributes-integer undef
13:44:36.450760 tramp-get-file-property (8) # /user/username/_darcs file-attributes-integer undef
13:44:36.481956 tramp-get-file-property (8) # /user/_darcs file-attributes-integer undef
13:44:36.483327 tramp-get-file-property (8) # /_darcs file-attributes-integer undef
13:44:36.486077 tramp-get-file-property (8) # /user/username/rebar.config file-attributes-integer undef
13:44:36.487299 tramp-get-file-property (8) # /user/username/project.clj file-attributes-integer undef
13:44:36.488504 tramp-get-file-property (8) # /user/username/build.boot file-attributes-integer undef
13:44:36.489690 tramp-get-file-property (8) # /user/username/SConstruct file-attributes-integer undef
13:44:36.490905 tramp-get-file-property (8) # /user/username/pom.xml file-attributes-integer undef
13:44:36.492121 tramp-get-file-property (8) # /user/username/build.sbt file-attributes-integer undef
13:44:36.493286 tramp-get-file-property (8) # /user/username/gradlew file-attributes-integer undef
13:44:36.494495 tramp-get-file-property (8) # /user/username/build.gradle file-attributes-integer undef
13:44:36.495776 tramp-get-file-property (8) # /user/username/Gemfile file-attributes-integer undef
13:44:36.497031 tramp-get-file-property (8) # /user/username/requirements.txt file-attributes-integer undef
13:44:36.498286 tramp-get-file-property (8) # /user/username/setup.py file-attributes-integer undef
13:44:36.499523 tramp-get-file-property (8) # /user/username/tox.ini file-attributes-integer undef
13:44:36.500783 tramp-get-file-property (8) # /user/username/package.json file-attributes-integer undef
13:44:36.502018 tramp-get-file-property (8) # /user/username/gulpfile.js file-attributes-integer undef
13:44:36.503223 tramp-get-file-property (8) # /user/username/Gruntfile.js file-attributes-integer undef
13:44:36.504536 tramp-get-file-property (8) # /user/username/bower.json file-attributes-integer undef
13:44:36.505994 tramp-get-file-property (8) # /user/username/composer.json file-attributes-integer undef
13:44:36.507247 tramp-get-file-property (8) # /user/username/Cargo.toml file-attributes-integer undef
13:44:36.508592 tramp-get-file-property (8) # /user/username/mix.exs file-attributes-integer undef
13:44:36.509832 tramp-get-file-property (8) # /user/username/stack.yaml file-attributes-integer undef
13:44:36.511123 tramp-get-file-property (8) # /user/rebar.config file-attributes-integer undef
13:44:36.512361 tramp-get-file-property (8) # /user/project.clj file-attributes-integer undef
13:44:36.513589 tramp-get-file-property (8) # /user/build.boot file-attributes-integer undef
13:44:36.514825 tramp-get-file-property (8) # /user/SConstruct file-attributes-integer undef
13:44:36.516035 tramp-get-file-property (8) # /user/pom.xml file-attributes-integer undef
13:44:36.517280 tramp-get-file-property (8) # /user/build.sbt file-attributes-integer undef
13:44:36.518474 tramp-get-file-property (8) # /user/gradlew file-attributes-integer undef
13:44:36.519773 tramp-get-file-property (8) # /user/build.gradle file-attributes-integer undef
13:44:36.521041 tramp-get-file-property (8) # /user/Gemfile file-attributes-integer undef
13:44:36.522334 tramp-get-file-property (8) # /user/requirements.txt file-attributes-integer undef
13:44:36.523599 tramp-get-file-property (8) # /user/setup.py file-attributes-integer undef
13:44:36.524846 tramp-get-file-property (8) # /user/tox.ini file-attributes-integer undef
13:44:36.526039 tramp-get-file-property (8) # /user/package.json file-attributes-integer undef
13:44:36.527322 tramp-get-file-property (8) # /user/gulpfile.js file-attributes-integer undef
13:44:36.528550 tramp-get-file-property (8) # /user/Gruntfile.js file-attributes-integer undef
13:44:36.529770 tramp-get-file-property (8) # /user/bower.json file-attributes-integer undef
13:44:36.530989 tramp-get-file-property (8) # /user/composer.json file-attributes-integer undef
13:44:36.532213 tramp-get-file-property (8) # /user/Cargo.toml file-attributes-integer undef
13:44:36.533389 tramp-get-file-property (8) # /user/mix.exs file-attributes-integer undef
13:44:36.534564 tramp-get-file-property (8) # /user/stack.yaml file-attributes-integer undef
13:44:36.535839 tramp-get-file-property (8) # /rebar.config file-attributes-integer undef
13:44:36.537110 tramp-get-file-property (8) # /project.clj file-attributes-integer undef
13:44:36.538282 tramp-get-file-property (8) # /build.boot file-attributes-integer undef
13:44:36.539475 tramp-get-file-property (8) # /SConstruct file-attributes-integer undef
13:44:36.540704 tramp-get-file-property (8) # /pom.xml file-attributes-integer undef
13:44:36.541887 tramp-get-file-property (8) # /build.sbt file-attributes-integer undef
13:44:36.543174 tramp-get-file-property (8) # /gradlew file-attributes-integer undef
13:44:36.544353 tramp-get-file-property (8) # /build.gradle file-attributes-integer undef
13:44:36.545525 tramp-get-file-property (8) # /Gemfile file-attributes-integer undef
13:44:36.546722 tramp-get-file-property (8) # /requirements.txt file-attributes-integer undef
13:44:36.585418 tramp-get-file-property (8) # /setup.py file-attributes-integer undef
13:44:36.586829 tramp-get-file-property (8) # /tox.ini file-attributes-integer undef
13:44:36.588065 tramp-get-file-property (8) # /package.json file-attributes-integer undef
13:44:36.589271 tramp-get-file-property (8) # /gulpfile.js file-attributes-integer undef
13:44:36.590468 tramp-get-file-property (8) # /Gruntfile.js file-attributes-integer undef
13:44:36.591684 tramp-get-file-property (8) # /bower.json file-attributes-integer undef
13:44:36.592881 tramp-get-file-property (8) # /composer.json file-attributes-integer undef
13:44:36.594046 tramp-get-file-property (8) # /Cargo.toml file-attributes-integer undef
13:44:36.595260 tramp-get-file-property (8) # /mix.exs file-attributes-integer undef
13:44:36.596456 tramp-get-file-property (8) # /stack.yaml file-attributes-integer undef
13:44:36.599156 tramp-get-file-property (8) # /user/username/.svn file-attributes-integer undef
13:44:36.600428 tramp-get-file-property (8) # /user/.svn file-attributes-integer undef
13:44:36.601697 tramp-get-file-property (8) # /.svn file-attributes-integer undef
13:44:36.603020 tramp-get-file-property (8) # /user/username/CVS file-attributes-integer undef
13:44:36.604284 tramp-get-file-property (8) # /user/CVS file-attributes-integer undef
13:44:36.605637 tramp-get-file-property (8) # /CVS file-attributes-integer undef
13:44:36.606914 tramp-get-file-property (8) # /user/username/Makefile file-attributes-integer undef
13:44:36.608227 tramp-get-file-property (8) # /user/Makefile file-attributes-integer undef
13:44:36.609496 tramp-get-file-property (8) # /Makefile file-attributes-integer undef
```

**Comment by raghavgautam** [comment]
I tried to reproduce the issue in my setup. This is an issue that I ran into which might be relevant to you.
I was not able to open the file and was getting this error:

```
error in process filter: open-network-stream: datanode-1/20075 nodename nor servname provided, or not known
```

The issue was that the ip for datanode-1 was not getting resolved. HDFS master redirect the request to one of the datanodes for fetching the data. Once I added the ips to hostname mapping to my /etc/hosts things were working fine.
Hope if it helps.
